### PR TITLE
Trim intervals loaded from interval files.

### PR DESCRIPTION
**Pull request by cmnbroad**
Fixes https://github.com/broadinstitute/gatk/issues/6371.

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/utils/IntervalUtilsUnitTest.java [hunk 1]
```diff
@@ -1120,6 +1120,22 @@ public void testUnmergedIntervals(String unmergedIntervals) {
         Assert.assertEquals(merged.size(), 1);
     }
 
+    @Test
+    public void testIntervalFileIntervalsAreTrimmed() throws Exception {
+        final GenomeLoc expectedGenomeLoc = hg19GenomeLocParser.createGenomeLoc("1", 1, 10);
+        final File gatkIntervalFile = createTempFile("testIntervalFileIntervalsAreTrimmed", ".intervals.list",
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/IntervalUtilsUnitTest.java]: Huh, I've never seen this method which creates the temp file and writes to it before.   That seems useful.  Maybe it should be promote somewhere.

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java [hunk 2]
```diff
@@ -403,8 +403,9 @@ else if ( glParser.isValidGenomeLoc(interval.getContig(), interval.getStart(), i
             } else {
                 try (PathLineIterator reader = new PathLineIterator(inputPath)) {
                     for (final String line : reader) {
-                        if (!line.trim().isEmpty()) {
-                            ret.add(glParser.parseGenomeLoc(line));
+                        final String trimmedLine = line.trim();
+                        if (!trimmedLine.isEmpty()) {
+                            ret.add(glParser.parseGenomeLoc(trimmedLine));
```
[the remark this answers is not in the document]
Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Added trim call to parserLoc, with a test.

File: src/test/java/org/broadinstitute/hellbender/utils/IntervalUtilsUnitTest.java (hunk 1 above)
Comment (cmnbroad) [reply src/test/java/org/broadinstitute/hellbender/utils/IntervalUtilsUnitTest.java]: I moved it to IOUtils with other methods with similar functionality, though now its no longer test-only.

**Comment by cmnbroad** [comment]
Back to you @lbergelson .

**Pull request merged**